### PR TITLE
Implement readSaveDetails, readScenarioOptions

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Wip

### CODE REVIEW

File: src/OpenLoco/src/S5/S5.cpp [hunk 1]
```diff
@@ -722,19 +722,54 @@ namespace OpenLoco::S5
         }
     }
 
+    // 0x00442403
+    std::unique_ptr<SaveDetails> peakSaveDetails(const fs::path& path)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/S5/S5.cpp:726]:
```suggestion
    std::unique_ptr<SaveDetails> peekSaveDetails(const fs::path& path)
```

**Comment by duncanspumpkin** [comment]
Do we think peek is a good name for this. It was a bit of a place holder.

**Comment by AaronVanGeffen** [comment]
I'd be fine with the names as-is, but @ZehMatt is right in that peeking is normally taken to mean reading ahead while not changing the referring file pointer. If that isn't the case here, perhaps we should change the names as suggested.

**Comment by IntelOrca** [comment]
I agree, peek would usually take a stream (and read something, but preserving the current position of the stream afterwards). Since you are taking a path and creating a file stream, you are just plain out reading data from a file and closing it.

File: src/OpenLoco/src/S5/S5.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #include "ScenarioObjective.h"
 #include <cstdint>
 #include <memory>
+#include <optional>
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/S5/S5.h:10]: Is this used? I only see `std::unique_ptr` as a new type in the diff.

Comment (duncanspumpkin) [reply src/OpenLoco/src/S5/S5.h:10]: no, good point.

**Pull request merged**